### PR TITLE
Replace old example URLs in `optuna.integrations`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

A sort of follow-up pull request of https://github.com/optuna/optuna/issues/2654. I found a few URLs are still old. This PR fixes them.

## Description of the changes

Replace old URLs.

### CODE REVIEW

File: optuna/integration/mxnet.py [hunk 1]
```diff
@@ -9,8 +9,8 @@
 class MXNetPruningCallback(object):
     """MXNet callback to prune unpromising trials.
 
-    See `the example <https://github.com/optuna/optuna/blob/master/
-    examples/mxnet/mxnet_integration.py>`__
+    See `the example <https://github.com/optuna/optuna-examples/blob/main/
+    mxnet/mxnet_simple.py>`__
```
Comment (toshihikoyanase) [inline optuna/integration/mxnet.py:13]: `mxnet_simple.py` does not use `MXNetPruningCallback`. Please revert the filename.
```suggestion
    mxnet/mxnet_integration.py>`__
```

File: optuna/integration/pytorch_lightning.py [hunk 2]
```diff
@@ -17,8 +17,8 @@
 class PyTorchLightningPruningCallback(Callback):
     """PyTorch Lightning callback to prune unpromising trials.
 
-    See `the example <https://github.com/optuna/optuna/blob/master/
-    examples/pytorch/pytorch_lightning_simple.py>`__
+    See `the example <https://github.com/optuna/optuna-examples/blob/
+    pytorch/pytorch_lightning_simple.py>`__
```
Comment (toshihikoyanase) [inline optuna/integration/pytorch_lightning.py:21]: The link is broken. I guess the branch name is missing.
```suggestion
    main/pytorch/pytorch_lightning_simple.py>`__
```

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
I'm not aware of the broken links. Thank you!

**Comment by nzw0301** [comment]
@toshihikoyanase thank you for your careful check!

**Comment by HideakiImamura** [comment]
@himkt Could you review this PR?

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Review by himkt — APPROVED** [review]
Nice catch!

**Pull request merged**